platform-sdk/consensus-otter-tests/src/testFixtures/java/org/hiero/otter/fixtures/turtle/TurtleNode.java

```diff
@@ -347,39 +288,31 @@ void destroy() throws InterruptedException {
         }
     }
 
-    private void checkLifeCycle(@NonNull final LifeCycle expected, @NonNull final String message) {
-        if (lifeCycle == expected) {
-            throw new IllegalStateException(message);
-        }
-    }
-
     private void doShutdownNode() throws InterruptedException {
-        if (lifeCycle == LifeCycle.STARTED) {
+        if (lifeCycle == RUNNING) {
             assert platform != null; // platform must be initialized if lifeCycle is STARTED
-            assert platformWiring != null; // platformWiring must be initialized if lifeCycle is STARTED
-            getMetricsProvider().removePlatformMetrics(platform.getSelfId());
-            platformWiring.stop();
-            platform.getNotificationEngine().unregisterAll();
+            platform.destroy();
             platformStatus = null;
             platform = null;
             platformWiring = null;
             model = null;
         }
-        lifeCycle = LifeCycle.SHUTDOWN;
+        lifeCycle = SHUTDOWN;
     }
 
     private void doStartNode() {
 
         final Configuration currentConfiguration = nodeConfiguration.createConfiguration();
+        final org.hiero.consensus.model.node.NodeId oldNodeId = org.hiero.consensus.model.node.NodeId.of(selfId.id());
```

Comment:
I find the name `oldNodeId` really confusing because I always expect that there is going to be a `newNodeId`. What do you think about calling this `legacyNodeId`?